src/metrics/index.ts

```diff
@@ -0,0 +1,55 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * An enum representing the possible metric lifetimes.
+ */
+export const enum Lifetime {
+  // The metric is reset with each sent ping
+  Ping = "ping",
+  // The metric is reset on application restart
+  Application = "application",
+  // The metric is reset with each user profile
+  User = "user",
+}
+
+/**
+ * The common set of data shared across all different metric types.
+ */
+interface CommonMetricData {
+  // The metric's name.
+  readonly name: string,
+  // The metric's category.
+  readonly category?: string,
+  // List of ping names to include this metric in.
+  readonly sendInPings: string[],
+  // The metric's lifetime.
+  readonly lifetime: Lifetime,
+  // Whether or not the metric is disabled.
+  //
+  // Disabled metrics are never recorded.
+  readonly disabled: boolean
+}
+
+class Metric implements CommonMetricData {
+  readonly type: string;
+  readonly name: string;
+  readonly category?: string;
+  readonly sendInPings: string[];
+  readonly lifetime: Lifetime;
+  readonly disabled: boolean;
+
+  constructor(type: string, meta: CommonMetricData) {
+    this.type = type;
+
+    this.name = meta.name;
+    this.sendInPings = meta.sendInPings;
+    this.lifetime = meta.lifetime;
+    this.disabled = meta.disabled;
+
+    if (meta.category) this.category = meta.category;
```

Comment:
Can we retain the same style that we used in the rest of the code?

```suggestion
    if (meta.category) {
      this.category = meta.category;
    }
```